bin/propolis-server/src/lib/server.rs

```diff
@@ -218,10 +230,18 @@ impl ServiceProviders {
             // Wait for the serial task to exit
             let _ = serial_task.task.await;
         }
-        if let Some(server) = self.oximeter_server_task.lock().await.take() {
+
+        // Clean up oximeter tasks and statistic state.
+        let mut oximeter_state = self.oximeter_state.lock().await;
+        if let Some((shutdown, task)) = oximeter_state.registration_task.take()
+        {
+            shutdown.notify_one();
```

Comment:
nit, take it or leave it: since this is used to signal shutdown only a single time, it might make more sense to use a `tokio::sync::oneshot` of `()`s? `Notify` is a bit more complex as it can be woken multiple times. not a big deal.

Reply:
Changed to a oneshot in 7c4cbe7